examples/nodejs/lambda-examples/cloudwatch-metrics/infrastructure/lib/momento-metrics-stack.ts

```diff
@@ -0,0 +1,310 @@
+      [exampleApp.Custom, customLogGroupName]
+    ])
+    const logGroupName = configToLogGroupName.get(stackConfig);
+
+    const logGroup = new LogGroup(this, 'Logs', {
+      logGroupName: logGroupName,
+      retention: RetentionDays.ONE_DAY,
+    });
+
+    switch (stackConfig) {
+      case exampleApp.NodejsLambda: {
+        const nodejsLambda = new lambdaNodejs.NodejsFunction(this, 'MomentoMetricsMiddlewareCDKExample', {
+          functionName: 'MomentoMetricsMiddlewareCDKExample',
+          runtime: lambda.Runtime.NODEJS_16_X,
+          entry: path.join(__dirname, '../../lambda/handler.ts'),
+          projectRoot: path.join(__dirname, '../../lambda'),
+          depsLockFilePath: path.join(__dirname, '../../lambda/package-lock.json'),
+          handler: 'handler',
+          timeout: cdk.Duration.minutes(6),
+          memorySize: 128,
+          environment: {
+            MOMENTO_API_KEY_SECRET_NAME: apiKeySecret.secretName,
+          },
+        });
+        apiKeySecret.grantRead(nodejsLambda);
+        break;
+      }
+      case exampleApp.NodejsEcs: {
+        const cluster = new ecs.Cluster(this, 'MomentoMetricsExampleFargateCluster');
+
+        const imageAsset = new DockerImageAsset(this, "MomentoMetricsECSDockerImage", {
+          directory: path.join(__dirname, "../../docker")
+        });
+
+        const taskDefinition = new ecs.FargateTaskDefinition(this, 'MomentoMetricsMiddlewareCDKExample');
+        taskDefinition.addContainer('MomentoMetricsECSContainer', {
+          image: ecs.ContainerImage.fromDockerImageAsset(imageAsset),
+          logging: new ecs.AwsLogDriver({
+            logGroup: logGroup,
+            streamPrefix: "MomentoMetricsMiddlewareCDKExample"
+          })
+        });
+        taskDefinition.addToTaskRolePolicy(new iam.PolicyStatement({
+          effect: iam.Effect.ALLOW,
+          actions: [
+            "secretsmanager:ListSecrets",
+            "secretsmanager:GetSecretValue",
+            "logs:PutLogEvents",
+            "logs:CreateLogStream",
+            "ecs:*"
+          ],
+          resources: ["*"],
+        }));
+        taskDefinition.addToExecutionRolePolicy(new iam.PolicyStatement({
+          effect: iam.Effect.ALLOW,
+          actions: [
+            "secretsmanager:ListSecrets",
+            "secretsmanager:GetSecretValue",
+            "logs:PutLogEvents",
+            "logs:CreateLogStream",
+            "ecs:*"
+          ],
+          resources: ["*"],
+        }));
+
+        new ecs.FargateService(this, 'MomentoMetricsECSFargateService', {
+          cluster,
+          taskDefinition,
+        });
+        break;
+      }
+      case exampleApp.Custom: {
+        console.log('Skipping to dashboard creation');
+        break;
+      }
+      default: {
+        throw new Error('Unimplemented CDK stack application');
+      }
+    }
+
+    logGroup.addMetricFilter('ExampleMetricFilterDuration', {
```

Comment:
would probably be useful to add a code comment above this part, explaining what it is doing. "The Momento experimental metrics middleware produces JSON logs with metrics about each Momento request.  An example log entry looks like this: ...

In this section we use CloudWatch Metric Filters (maybe link to docs) to extract metrics from these log messages."